src/client/src/tscSQLParser.c

```diff
@@ -1657,8 +1657,8 @@ static bool validateTagParams(SArray* pTagsList, SArray* pFieldList, SSqlCmd* pC
   // field name must be unique
   for (int32_t i = 0; i < numOfTags; ++i) {
     TAOS_FIELD* p = taosArrayGet(pTagsList, i);
```

Comment:
Check name against the slice [0, numOfCol) of pFieldList with numOfCols exclusive.